File: hw/bonilla/u2/hw19Inheritance/FarmSystemJavaScript/view/FarmSystem.js
const Cage = require('../model/Cage');
const Chicken = require('../model/Chicken');
const Cow = require('../model/Cow');
const Pig = require('../model/Pig');
const Sheep = require('../model/Sheep');
const Location = require('../model/Location');

class CustomLocation extends Location {
    constructor(xCoordinate, yCoordinate) {
        super(xCoordinate, yCoordinate);
    }
}

function main() {
    let id;
    let breed;
    let bornOn;
    let gender;
    let isAbleToReproduce;
    let weight;
    let cage;
    let location;
    let xCoordinate;
    let yCoordinate;

    xCoordinate = 10;
    yCoordinate = 20;
    weight = 10.4;
    gender = "male";
    isAbleToReproduce = false;

    bornOn = new Date(2025, 2, 1); 
    breed = "Holstein";

    location = new CustomLocation(xCoordinate, yCoordinate);

    cage = new Cage(1, "Stable for cows", 2, location);

    let farmAnimal;

    farmAnimal = new Chicken(true, 0, 1, breed, bornOn, gender, true, weight, cage);
    console.log("farmAnimal --> Chicken " + farmAnimal.toString());

    farmAnimal = new Pig(2, breed, bornOn, gender, false, weight, cage);
    console.log("farmAnimal --> Pig " + farmAnimal.toString());

    farmAnimal = new Cow(false, weight, 3, breed, bornOn, gender, true, weight, cage);
    console.log("farmAnimal --> Cow " + farmAnimal.toString());

    farmAnimal = new Sheep(bornOn, 4, breed, bornOn, gender, false, weight, cage);
    console.log("farmAnimal --> Sheep " + farmAnimal.toString());
}

main();

File: hw/carvajal/u2/hw19Inheritance/PythonFarmSystem/view/farm_system.py
from datetime import datetime
from model.cage import Cage
from model.location import Location
from model.chicken import Chicken
from model.pig import Pig
from model.cow import Cow
from model.sheep import Sheep

def main():
    x_coordinate = 10
    y_coordinate = 20
    weight = 10.4
    gender = "male"
    is_able_to_reproduce = False

    date = datetime(2025, 3, 1)
    breed = "Holstein"

    location = Location(x_coordinate, y_coordinate)
    cage = Cage(1, "STABLE FOR COWS", 2, location)

    farm_animal = Chicken(True, 0, 1, breed, date, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --->", farm_animal)

    farm_animal = Pig(2, breed, date, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --->", farm_animal)

    farm_animal = Cow(True, 5, 3, breed, date, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --->", farm_animal)

    farm_animal = Sheep(date, 5, breed, date, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --->", farm_animal)

if __name__ == "__main__":
    main()
